### PR TITLE
Handle keybindings and confirmation in the indicator

### PR DESCRIPTION
**Pull request by lenemter**
Fixes https://github.com/elementary/wingpanel-indicator-sound/issues/9
Fixes https://github.com/elementary/wingpanel-indicator-sound/issues/13

Todo:
- [x] Make shortcuts customizable

### CODE REVIEW

**Review by danirabbit — APPROVED** [review]
I can confirm this fixes the issue! Requested @Marukesu since it seems like he had some opinions on previous branches :)

File: src/Indicator.vala [hunk 1]
```diff
@@ -32,6 +32,12 @@ public class Sound.Indicator : Wingpanel.Indicator {
     private Notify.Notification? notification;
     private Services.VolumeControlPulse volume_control;
 
+    private ShellKeyGrabber? key_grabber = null;
+    private ulong key_grabber_id = 0;
+    private GLib.GenericSet<uint> volume_up_action_ids = new GLib.GenericSet<uint> (null, null);
+    private GLib.GenericSet<uint> volume_down_action_ids = new GLib.GenericSet<uint> (null, null);
+    private GLib.GenericSet<uint> volume_mute_action_ids = new GLib.GenericSet<uint> (null, null);
+
```
Comment (Marukesu) [inline src/Indicator.vala:40]: You could use a Gee.MultiMap, so there's less code to update here later if we decide to handle more keybindings.

**Review by Marukesu — APPROVED** [review]
LGTM. Don't forget to update the switchboard plug to use the new keys instead of the GSD ones.

File: src/Indicator.vala (hunk 1 above)
Comment (lenemter) [reply src/Indicator.vala]: Done

**Comment by lenemter** [comment]
I noticed the grabs are not updated when gsettings keys change. I'll fix this.

**Pull request merged**